### PR TITLE
Rework IsOneOf validator

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: shared/validate/validate.go [hunk 1]
```diff
@@ -107,12 +107,14 @@ func IsBool(value string) error {
 }
 
 // IsOneOf checks whether the string is present in the supplied slice of strings.
-func IsOneOf(value string, valid []string) error {
-	if !stringInSlice(value, valid) {
-		return fmt.Errorf("Invalid value %q (not one of %s)", value, valid)
-	}
+func IsOneOf(valid []string) func(value string) error {
```
Comment (tomponline) [inline shared/validate/validate.go:110]: Nice!

**Review by tomponline — APPROVED** [review]

File: shared/validate/validate.go (hunk 1 above)
Comment (tomponline) [inline shared/validate/validate.go:110]: @stgraber although we can go further now, and make this a variadic.

So it becomes:

```
"limits.memory.enforce": validate.Optional(validate.IsOneOf("soft", "hard")),
```

Comment (stgraber) [reply shared/validate/validate.go:110]: Oh yeah, we can do that, that'll be another fun change to do throughout the code base, but makes sense :)

Comment (tomponline) [reply shared/validate/validate.go:110]: Thanks, and the nice thing about variadics is if you happen to already have a slice of valid options you can still pass them just using the `...` suffix.

**Comment by tomponline** [comment]
Thanks!

**Pull request merged**